### PR TITLE
Small fix of vertex map

### PR DESCRIPTION
**Pull request by songqing**
## What do these changes do?

When running the test of vertex map, I encountered a small problem with the script, see
g1
Besides, some small code optimization of vertex map is submitted.

## Related issue number

Fixes

### CODE REVIEW

**Comment by RHINO-CI-Robot-enabled-with-GPT** [comment]
**[AI Review]** This comment is generated by an AI model (GPT-3.5-turbo).

Based on the provided patch, it seems that the changes are related to the VertexMap in the Grape library. The changes include modifying the way the vertices are mapped and converting the vertex ids with the help of mappings. Additionally, there are changes in the way the threads are created for sending and receiving messages and modifying the way the global and local vertex maps are built.

Without more context on the purpose of these changes, it is difficult to determine if they are correct or not. However, the changes seem to be consistent with the functionality of the Grape library. Therefore, it is recommended to review the purpose of these changes and test them thoroughly before merging the pull request.

File: grape/vertex_map/local_vertex_map.h [hunk 1]
```diff
@@ -89,13 +89,13 @@ class LocalVertexMapBuilder {
...
         sync_comm::Recv(keys, src_worker_id, 0, comm_spec.comm());
-        std::vector<VID_T> gid_list;
+        std::vector<VID_T> gid_list(keys.size());
         VID_T gid;
         auto& native_indexer = oid_to_index_[fid_];
         for (size_t k = 0; k < keys.size(); ++k) {
           CHECK(native_indexer.get_index(keys[k], gid));
           gid = id_parser_.generate_global_id(fid_, gid);
-          gid_list.push_back(gid);
+          gid_list[k] = gid;
```
Comment (sighingnow) [inline grape/vertex_map/local_vertex_map.h:98]: Nice catch!

File: grape/vertex_map/global_vertex_map.h [hunk 2]
```diff
@@ -70,30 +70,19 @@ class GlobalVertexMapBuilder {
...
         int dst_worker_id = (worker_id + worker_num - 1) % worker_num;
         while (dst_worker_id != worker_id) {
-          for (fid_t fid = 0; fid < fnum; ++fid) {
-            if (comm_spec.FragToWorker(fid) != worker_id) {
-              continue;
-            }
-            sync_comm::Send(indexer_, dst_worker_id, 0, comm_spec.comm());
-          }
+          sync_comm::Send(indexer_, dst_worker_id, 0, comm_spec.comm());
           dst_worker_id = (dst_worker_id + worker_num - 1) % worker_num;
```
Comment (sighingnow) [inline grape/vertex_map/global_vertex_map.h:86]: `comm_spec.FragToWorker(fid) == worker_id` may not always holds if one worker has more than one fragments (it is not the case in libgrape-lite itself but might be implemented in the future).

So please keep this part as it was.

File: tests/vertex_map_tests.cc [hunk 3]
```diff
@@ -25,15 +33,6 @@ limitations under the License.
...
 
-#include <grape/fragment/immutable_edgecut_fragment.h>
-#include <grape/fragment/loader.h>
-#include <grape/fragment/mutable_edgecut_fragment.h>
-#include <grape/fragment/partitioner.h>
-#include <grape/grape.h>
-#include <grape/util.h>
-#include <grape/vertex_map/global_vertex_map.h>
-#include <grape/vertex_map/local_vertex_map.h>
-
```
Comment (sighingnow) [inline tests/vertex_map_tests.cc:36]: Don't touch this part, the self headers should be placed after std headers, the original behaviour should be ok.

File: tests/vertex_map_tests.cc [hunk 4]
```diff
@@ -218,16 +217,11 @@ void Run() {
...
   // FIXME: no barrier apps. more manager? or use a dynamic-cast.
-  std::string efile = FLAGS_efile;
-  std::string vfile = FLAGS_vfile;
-  std::string delta_efile = FLAGS_delta_efile;
-  std::string delta_vfile = FLAGS_delta_vfile;
-  std::string out_prefix = FLAGS_out_prefix;
   auto spec = grape::MultiProcessSpec(comm_spec, __AFFINITY__);
   int fnum = comm_spec.fnum();
-  CreateAndQuery<OID_T, VID_T, VDATA_T, double, grape::LoadStrategy::kOnlyOut,
+  CreateAndQuery<OID_T, VID_T, VDATA_T, EDATA_T, grape::LoadStrategy::kOnlyOut,
```
Comment (sighingnow) [inline tests/vertex_map_tests.cc:222]: Nice catch.

**Review by sighingnow — COMMENTED** [review]
Basically LGTM, except some blocks that shoudn't be changed.

@songqing could you please take a minor revision on this pull request so we can get it merged? Thanks!

File: tests/vertex_map_tests.cc (hunk 3 above)
Comment (songqing) [reply tests/vertex_map_tests.cc:36]: It's done by clang-format automatically, in other files of this repo, the include order is same with the changed by me, you can see this file [run_app.h](https://github.com/alibaba/libgrape-lite/blob/master/examples/analytical_apps/run_app.h).
Or, I can change the <> to ""

File: grape/vertex_map/global_vertex_map.h (hunk 2 above)
Comment (songqing) [reply grape/vertex_map/global_vertex_map.h]: done

File: tests/vertex_map_tests.cc (hunk 3 above)
Comment (sighingnow) [reply tests/vertex_map_tests.cc:36]: Ok. We could get this merged and I will tweak the clang-format configuration.

Thanks!

**Pull request merged**